### PR TITLE
Enhancement: Improve Documentation & Yoda Condition

### PR DESCRIPTION
**Pull request by ahmadawais**
This PR improves the documentation as per the WordPress Documentation Standards. It also addresses a Yoda Condition error.

Replaces #278 h/t @aduth and @youknowriad 
Looking forward!

### CODE REVIEW

**Review by youknowriad — APPROVED** [review]
This LGTM but I'll leave other reviewers check cause I'm not an expert in WordPress Plugins and WordPress Plugins Documentation :)

**Comment by nylen** [comment]
Previously #278.  I have similar feedback to what @aduth said over there.

File: index.php [hunk 1]
```diff
@@ -1,14 +1,21 @@
 <?php
-
 /**
  * Plugin Name: Gutenberg
  * Plugin URI: https://wordpress.github.io/gutenberg/
  * Description: Prototyping since 1440. Development plugin for the editor focus in core.
  * Version: 0.1
+ *
+ * @package WordPress
```
Comment (nylen) [inline index.php]: I think we should not add this unless/until this code ships in core.

File: index.php [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
  */
 
+// Hook.
```
Comment (nylen) [inline index.php]: -1 on this comment and all others that just say "Hook", it doesn't add value.

File: index.php [hunk 3]
```diff
@@ -19,13 +26,26 @@ function gutenberg_menu() {
...
+add_action( 'admin_enqueue_scripts', 'gutenberg_scripts_and_styles' );
+
+/**
+ * Scripts & Styles.
+ *
+ * @param  string $hook Screen name.
+ * @since  4.8.0
+ */
```
Comment (nylen) [inline index.php]: Also needs a brief explanation.

> Enqueues scripts and styles when visiting the top-level page of the Gutenberg editor.

File: index.php [hunk 4]
```diff
@@ -1,14 +1,21 @@
...
  */
 
+// Hook.
 add_action( 'admin_menu', 'gutenberg_menu' );
 
+/**
+ * Gutenberg's Menu.
+ *
+ * @since  4.8.0
+ */
```
Comment (nylen) [inline index.php]: This block should describe (even if just in a short sentence) what the function actually *does*.

> Adds a new menu page for the Gutenberg editor.

File: index.php [hunk 5]
```diff
@@ -19,13 +26,26 @@ function gutenberg_menu() {
...
 		wp_enqueue_script( 'gutenberg_js', plugins_url( 'build/app.js', __FILE__ ) );
 	}
 }
-add_action( 'admin_enqueue_scripts', 'gutenberg_scripts_and_styles' );
 
+/**
+ * Project.
+ *
+ * @since  4.8.0
+ */
```
Comment (nylen) [inline index.php]: > The main entry point for the Gutenberg editor.  Renders the editor on the wp-admin page for the plugin.

**Review by nylen — CHANGES_REQUESTED** [review]
Adding inline documentation is a good idea, but we need to make sure it adds value rather than just stubs without much content.

File: index.php (hunk 2 above)
Comment (ahmadawais) [inline index.php]: @nylen Do you have a suggestion to what should go here? Leaving it without inline docs, feels weird. I appreciate your feedback.

File: index.php (hunk 1 above)
Comment (ahmadawais) [inline index.php]: Sure, I can remove this for now.

File: index.php (hunk 2 above)
Comment (nylen) [inline index.php]: I would leave it without docs, yes.  I would also move the calls immediately below the functions where they are defined.  This combination of inline docblock, function code, `add_action` is enough to be very clear, IMO.

WP core only requires documentation for the `do_action` or `apply_filters` calls, not the places where the code actually _uses_ a filter ([ref](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#4-hooks-actions-and-filters)).

**Comment by ahmadawais** [comment]
@nylen Thanks for your feedback. I have made the changes, take a look when you are up for it.

**Comment by ahmadawais** [comment]
@nylen Take a look now.

**Review by nylen — APPROVED** [review]
I made some very small formatting tweaks.  This looks good to me, thanks for doing it :)

**Comment by ahmadawais** [comment]
@nylen Thanks for the review. I believe that adding more than 80 characters in a single line of `long description` is a bad practice which is why I split the long desc at two lines.

Anywho, thanks! I am on it and will try to contribute more.

**Comment by nylen** [comment]
> I believe that adding more than 80 characters in a single line of `long description` is a bad practice which is why I split the long desc at two lines.

I agree!  I'm not a fan of long lines in code either.

I used my editor to wrap these lines at 80 chars, rather than breaking at the end of a sentence.  If you see one that I missed, let me know and we'll fix it :)

**Pull request merged**